File: ClientTestLibrary/TestSystemFunctions.cs
using System;
using Bridge;
using Bridge.QUnit1;

namespace ClientTestLibrary
{
    // Tests DateTime functions
    [FileName("testSets.js")]
    class TestSystemFunctions
    {
        // DateTime functions
        public static void Test1(Assert assert)
        {
            assert.Expect(2);

            //[#83] by C#
            var str = "2015-03-24T10:48:09.1500225+03:00";
            var bridgeDate = DateTime.Parse(str);
            var bridgeDate1 = new DateTime(str);
            assert.DeepEqual(bridgeDate, bridgeDate1, "[#83] C# bridgeDate = bridgeDate1");

            //[#83] by js
            Script.Write<dynamic>(@"var str = ""2015-03-24T10:48:09.1500225+03:00"",
    bridgeDate = Bridge.Date.parse(str),
    jsDate = new Date(Date.parse(str)),
    format = ""yyyy-MM-dd hh:mm:ss"";
    deepEqual(Bridge.Date.format(bridgeDate, format), Bridge.Date.format(jsDate, format), ""[#83] js"")");
        }

        // String functions
        public static void Test2(Assert assert)
        {
            assert.Expect(2);

            var msg = "HELLO".ToLowerCase();
            assert.DeepEqual(msg, "hello", "'HELLO'.ToLowerCase()");

            msg = "hello".ToUpperCase();
            assert.DeepEqual(msg, "HELLO", "'hello'.ToUpperCase()");
        }
    }
}

File: www/js/testSystemFunctions.js
Bridge.define('ClientTestLibrary.TestSystemFunctions', {
    statics: {
        dateTimes: function (assert) {
            assert.expect(2);

            // TEST
            //[#83] by C#
            var str = "2015-03-24T10:48:09.1500225+03:00";
            var bridgeDate = Bridge.Date.parse(str);
            var bridgeDate1 = new Date(str);

            assert.deepEqual(bridgeDate, bridgeDate1, "[#83] C# bridgeDate = bridgeDate1");

            // TEST
            //[#83] by JavaScript code. This is to check the same issue as above and just to check another way of calling QUnit from JavaScript
            var str = "2015-03-24T10:48:09.1500225+03:00",
    bridgeDate = Bridge.Date.parse(str),
    jsDate = new Date(Date.parse(str)),
    format = "yyyy-MM-dd hh:mm:ss";
    assert.deepEqual(Bridge.Date.format(bridgeDate, format), Bridge.Date.format(jsDate, format), "[#83] js");
        },
        strings: function (assert) {
            assert.expect(2);

            // TEST
            var msg = "HELLO".toLowerCase();
            assert.deepEqual(msg, "hello", "'HELLO'.ToLowerCase()");

            // TEST
            msg = "hello".toUpperCase();
            assert.deepEqual(msg, "HELLO", "'hello'.ToUpperCase()");
        },
        stringBuilders: function (assert) {
            assert.expect(1);

            // TEST
            var sb = new Bridge.Text.StringBuilder("foo");

            assert.deepEqual(sb.toString(), "foo", "StringBuilder(string) .ctor");
        }
    }
});

